tests/kernel/interrupt/prj.conf

```diff
@@ -4,3 +4,4 @@ CONFIG_DYNAMIC_INTERRUPTS=y
 CONFIG_MP_MAX_NUM_CPUS=1
 CONFIG_THREAD_STACK_INFO=y
 CONFIG_IRQ_OFFLOAD=y
+CONFIG_ISR_SHARE_IRQ=y
```

Comment:
please keep the existing test cases to test what they already do today, that is without introducing `CONFIG_ISR_SHARE_IRQ=y`.

Instead add a new test case here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/kernel/interrupt/testcase.yaml
For example: `arch.interrupt.shared:`

where you can set `CONFIG_ISR_SHARE_IRQ=y`.

That will also allow you to limit testing to archs you know are supported.